gadget/validate.go

```diff
@@ -230,31 +240,73 @@ func ensureRolesConsistency(roles map[string]*roleInstance, expectedSeed bool) e
 	return nil
 }
 
+func ensureRolesConsistencyOnModalClassic(roles map[string]*roleInstance) error {
+	if roles[SystemBoot] == nil || roles[SystemData] == nil {
+		return fmt.Errorf("system-boot and system-data roles are needed on classic")
+	}
+
+	// Make sure labels are as expected - save is optional
+	roleLabelToCheck := []roleLabel{
+		{SystemBoot, ubuntuBootLabel}, {SystemData, ubuntuDataLabel}}
+	roleLabelOptional := []roleLabel{
+		{SystemSeed, ubuntuSeedLabel}, {SystemSave, ubuntuSaveLabel}}
+	for _, rlLb := range roleLabelOptional {
+		if roles[rlLb.role] != nil {
+			roleLabelToCheck = append(roleLabelToCheck, rlLb)
+		}
+	}
+	if err := checkImplicitLabels(roleLabelToCheck, roles); err != nil {
+		return err
+	}
+
+	// Check that boot/seed/save are in the same volume
+	bootVolName := roles[SystemBoot].volName
+	for _, rlLb := range roleLabelOptional {
+		if roles[rlLb.role] != nil {
+			if roles[rlLb.role].volName != bootVolName {
```

Comment:
nitpick: these could be `and`-ed together

Reply:
Good catch, will fix that